src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockDepths/boss_magmus.cpp

```diff
@@ -110,7 +117,61 @@ class boss_magmus : public CreatureScript
         }
 };
 
+class npc_ironhand_guardian : public CreatureScript
+{
+    public:
+
+        npc_ironhand_guardian() : CreatureScript("npc_ironhand_guardian") { }
+
+        struct npc_ironhand_guardianAI : public ScriptedAI
+        {
+
+            npc_ironhand_guardianAI(Creature* creature) : ScriptedAI(creature)
+            {
+                _instance = me->GetInstanceScript();
+            }
+
+            void Reset() override
+            {
+                _events.Reset();
+                _lastState = 0;
+            }
+
+            void UpdateAI(uint32 uiDiff) override
+            {
+                if (_lastState == 0)
+                {
+                    if (_instance->GetData(TYPE_IRON_HALL) == NOT_STARTED)
+                        return;
+                    _events.ScheduleEvent(EVENT_GOUTOFFLAME, 0, 0, 0);
```

Comment:
if will executed instant, why you need event to do it?

Reply:
It is probably because I'm not well versed in the entire event system (i.e. how to do it the correct way). The though was to separate the activation and the actual event.

If the Re-schedule below is changed to `Repeat`, would you prefer a `DoCastAOE` here together with the `ScheduleEvent(16s)` or to keep a zero-timed event and let the execute below handle the first cast?

Reply:
hmm it need repeat...
maybe just: `_events.ScheduleEvent(EVENT_GOUTOFFLAME, 0s);`
And use repeat inside of event handle